src/store/download/slice.ts

```diff
@@ -0,0 +1,52 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { FetchBlobResponse, StatefulPromise } from 'rn-fetch-blob'
+
+export type DownloadState = typeof initialState
+
+type State = {
+  downloadedPercentage: number
+  fetchTask: StatefulPromise<FetchBlobResponse>
```

Comment:
Generally only plain js objects should be stored in redux. This is to allow for things like time travel debugging and serializing the state. Is there any way you could store this fetch task elsewhere? Could just be a mutable variable in the handler file I think

Reply:
Hmm so the problem is that it needs to be accessed by the DownloadTrackProgressDrawer. I don't think putting it in the handler files would let DownloadTrackProgressDrawer access it. And I don't think we can pass the fetchTask object to the DownloadTrackProgressDrawer any other way. The only alternative I can think of, without putting it in the redux store is to make it a globally accessible variable but that doesn't seem like a great choice either. Happy to discuss further!

Reply:
an alternative here that would be more canonical would be to expose another action that the component could call to find & cancel the ongoing fetch task. that would let you track the fetch task entirely within the message handler/saga layer and not bleed into the store/component. 

A comment here about why we're doing this would also be reasonable, but i don't think this change would be too hard.

Reply:
Updated to pass the "cancel function" instead of the stateful promise!